internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/util/container"
+)
+
+const (
+	defaultIndexImage        = "quay.io/operator-framework/upstream-opm-builder:latest"
+	defaultContainerName     = "registry-grpc"
+	defaultContainerPortName = "grpc"
+	defaultGRPCPort          = 50051
+	semverBundleAddMode      = "semver"
+	replacesBundleAddMode    = "replaces"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	// BundleAddMode specifies the graph update mode that defines how channel graphs are updated
+	BundleAddMode string
+
+	// BundleImage specifies the container image that opm uses to generate and incrementally update the database
+	BundleImage string
+
+	// Index image contains a database of pointers to operator manifest content that is queriable via an API.
+	// new version of an operator bundle when published can be added to an index image
+	IndexImage string
+
+	// DBPath refers to the registry DB;
+	// if an index image is provided, the existing registry DB is located at /database/index.db
+	DBPath string
+
+	// Namespace refers to the specific namespace in which the registry pod will be created and scoped to
+	Namespace string
+
+	// Kubeclient refers to a Kubernetes clientset that implements kubernetes.Interface.
+	Kubeclient kubernetes.Interface
+
+	// GRPCPort is the container grpc port which is defaulted to 50051
+	GRPCPort int32
+}
+
+// newRegistryPod initializes the RegistryPod struct and sets defaults for empty fields
+func newRegistryPod(kubeclient kubernetes.Interface, dbPath, bundleImage, namespace string) *RegistryPod {
```

Comment:
If this function is only used in tests, move it to the test suite.

Reply:
@estroz acctually it should be called in the CreateOnCluster to create the object with the defaults.

Reply:
IMO this func need to be export. 
```suggestion
func NewWith(kubeclient kubernetes.Interface, dbPath, bundleImage, namespace string) *RegistryPod {
```

Then, who will consume it will use it as : 

```


```

Reply:
is it make more sense now @estroz @rashmigottipati ?

Reply:
I'd prefer `NewRegistryPod` if this is intended to be used externally.

Reply:
we will not use it in the CLI or in any place at some point? because now it is not called in any place. 
I am ok with `NewRegistryPod ` 👍

Reply:
We will be using it in the CLI. Created it as `NewRegistryPod`